### PR TITLE
updates in ApiConnection, WikibaseDataEditor and in WbEditingAction c…

### PR DESCRIPTION
**Pull request by chetansharmaatgithub**
Updated code to achieve single login and bot user access to perform actions at wiki server.

### CODE REVIEW

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java [hunk 1]
```diff
@@ -736,7 +739,9 @@ private JsonNode performAPIAction(
 			return null;
 		}
 
-		checkEditSpeed();
+		if (!bot) {
+			checkEditSpeed();
+		}
```
Comment (wetneb) [inline wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java]: The edit speed should be controlled even for bots. If you want to disable it, just configure the `WikibaseDataEditor` for a high editing speed: http://wikidata.github.io/Wikidata-Toolkit/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.html#setAverageTimePerEdit(int)

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java [hunk 2]
```diff
@@ -36,6 +36,28 @@
  */
 public class WikibaseDataEditor {
 
+	/**
+	 * Field to store current csrfToken.
+	 */
+	static String csrfToken;
```
Comment (wetneb) [inline wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java]: Same issue here - no reason to have this static. The CSRF token is already stored in the `WbEditingAction` instance, why do you need to duplicate it here?

**Comment by wetneb** [comment]
If you want to ensure a single login, you can do this in your own code by holding the `ApiConnection` in a singleton. This is not something we want to have in WDTK itself.

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java [hunk 3]
```diff
@@ -195,12 +195,12 @@
...
 	 * Map of cookies that are currently set.
 	 */
-	final Map<String, String> cookies;
+	static Map<String, String> cookies;
 
 	/**
 	 * Map of requested tokens.
 	 */
-	final Map<String, String> tokens;
+	static Map<String, String> tokens;
```
[the remark this answers is not in the document]
Comment (chetansharmaatgithub) [reply wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java]: ok i got it.

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java [hunk 4]
```diff
@@ -354,11 +358,38 @@ public String getCurrentUser() {
...
+	}
+
+	/**
+	 * Set user password value
+	 *
+	 * @param password
+	 */
+	public void setCurrentUserPassword(String password) {
+		this.password = password;
+	}
```
[the remark this answers is not in the document]
Comment (chetansharmaatgithub) [reply wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java]: ok i got it.

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java [hunk 5]
```diff
@@ -710,7 +710,7 @@ private JsonNode performAPIAction(
 		}
 		
 		if (bot) {
-			parameters.put("bot", "");
+			parameters.put("bot", "true");
```
[the remark this answers is not in the document]
Comment (chetansharmaatgithub) [reply wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java]: Now it is working fine even if i am not setting "true" value for bot param, you were right.

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java (hunk 1 above)
Comment (chetansharmaatgithub) [reply wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java]: Thanks, it's working.

File: wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java (hunk 2 above)
Comment (chetansharmaatgithub) [reply wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java]: ok i got it.

**Comment by chetansharmaatgithub** [comment]
Thanks for suggestions, I updated my code according your suggestions and got the desired results.

**Comment by wetneb** [comment]
Great! Closing this then.

**Pull request closed without merging**